features/tests/features/generators/test_datetime.py

```diff
@@ -92,3 +92,44 @@ def test_datetime_feature_generator(generator_helper, data_helper):
     assert list(output_data_2['datetime'].values) == list(output_data_2['datetime_as_object'].values)
     assert expected_output_data_feat_datetime == list(output_data_2['datetime'].values)
     assert expected_output_data_feat_datetime_hour == list(output_data_2['datetime.hour'].values)
+
+
+# This covers the nightmare input scenario for a datetime column:
+# multiple formats, multiple NaN's of different types, multiple time zones (including no time zone), all as strings.
+# This is just about as bad as it could get. If we work here, we should work with practically anything.
+def test_datetime_feature_generator_advanced(generator_helper, data_helper):
+    # Given
+    input_data = data_helper.generate_datetime_as_object_feature_advanced().to_frame(name='datetime_as_object')
+
+    generator = DatetimeFeatureGenerator()
+
+    expected_feature_metadata_in_full = {
+        ('object', ('datetime_as_object',)): ['datetime_as_object'],
+    }
+
+    expected_feature_metadata_full = {('int', ('datetime_as_int',)): [
+        'datetime_as_object',
+        'datetime_as_object.year',
+        'datetime_as_object.month',
+        'datetime_as_object.day',
+        'datetime_as_object.dayofweek'
+    ]}
+
+    expected_output_data_feat_datetime = [
+        1533140820000000000,
+        1600055337034500096,
+        1600055337034500096,
+        1628528828659000000,
+        1628528895541000000,
+        1610022803938000000
```

Comment:
This is not correct because time the known zones were stripped.
```
0    1533140820000000000
1    1600067037034500096
2    1600067037034500096
3    1628543228659000000
4    1628543295541000000
5    1610040803938000000
```